FinSearchUnified/Tests/Bundle/SearchBundleFindologic/ConditionHandler/ProductAttributeConditionHandlerTest.php

```diff
@@ -143,4 +144,33 @@ public function testGenerateCondition($operator, array $attributes, array $expec
             );
         }
     }
+
+    /**
+     * Data from Shopware's Criteria class already contains the query parameter URL decoded. Decoding them again
+     * must not be done! This test ensures that these information is never encoded multiple times.
+     */
+    public function testGenerateConditionDoesNotDecodeAlreadyDecodedData()
+    {
+        $filterName = 'blub';
+        $filterValue = 'this +"a_;# *<, +';
+        $config = new Config();
+        $config->setServiceId('ABCDABCDABCDABCDABCDABCDABCDABCD');
+
+        $this->handler->generateCondition(
+            new ProductAttributeCondition($filterName, Operator::EQ, $filterValue),
+            $this->querybuilder,
+            $this->context
+        );
+
+        $params = $this->querybuilder->getParameters();
+
+        static::assertArrayHasKey('attrib', $params);
+        static::assertArrayHasKey($filterName, $params['attrib']);
+        static::assertSame($filterValue, $params['attrib'][$filterName]['']);
```

Comment:
Why is there `['']`?

Reply:
This is due to how the URL structure of Findologic is. It is always `attrib[filterName][]=filterValue`, therefore `''` is just accessing that empty string array key. For a range slide for example it would be `attrib[filterName][min/max]=filterValue` and instead of `''` you would get the data from `'min'`/`'max'`.